gitops/snapshot.go

```diff
@@ -1158,3 +1158,19 @@ func GetSourceRepoOwnerFromSnapshot(snapshot *applicationapiv1alpha1.Snapshot) s
 	}
 	return ""
 }
+
+// GetShaFromSnapshot returns the value of "pac.test.appstudio.openshift.io/sha"
+// annotation of length 7 for short SHA from the Snapshot, if it exists.
+// Or else, returns an empty string.
+func GetShaFromSnapshot(ctx context.Context, snapshot *applicationapiv1alpha1.Snapshot) string {
+	log := log.FromContext(ctx)
+
+	sha, found := snapshot.GetAnnotations()[PipelineAsCodeSHAAnnotation]
+	if found {
+		return sha[:7]
```

Comment:
suggestion: maybe we could make sure to handle a case where there's a value that's shorter than 7 characters here just so we don't error out unexpectedly here.

Reply:
good point, added the checks with test coverage. PTAL.

Reply:
Thanks!